src/scripts/site.js

```diff
@@ -1213,103 +1619,314 @@ var Dashboard = {
-                if ("settings" === item) return "mypreferencesmenu.html";
-                if ("wizard" === item) return "wizardstart.html";
-                if ("manageserver" === item) return "dashboard.html";
-                if ("recordedtv" === item) return "livetv.html?tab=3&serverId=" + options.serverId;
-                if ("nextup" === item) return "list/list.html?type=nextup&serverId=" + options.serverId;
+
+            appRouter.showLocalLogin = function (serverId, manualLogin) {
+                Dashboard.navigate("login.html?serverid=" + serverId);
+            };
+
+            appRouter.showVideoOsd = function () {
+                return Dashboard.navigate("videoosd.html");
+            };
+
+            appRouter.showSelectServer = function () {
+                Dashboard.navigate(AppInfo.isNativeApp ? "selectserver.html" : "login.html");
+            };
+
+            appRouter.showWelcome = function () {
+                Dashboard.navigate(AppInfo.isNativeApp ? "selectserver.html" : "login.html");
+            };
+
+            appRouter.showSettings = function () {
+                Dashboard.navigate("mypreferencesmenu.html");
+            };
+
+            appRouter.showGuide = function () {
+                Dashboard.navigate("livetv.html?tab=1");
+            };
+
+            appRouter.goHome = function () {
+                Dashboard.navigate("home.html");
+            };
+
+            appRouter.showSearch = function () {
+                Dashboard.navigate("search.html");
+            };
+
+            appRouter.showLiveTV = function () {
+                Dashboard.navigate("livetv.html");
+            };
+
+            appRouter.showRecordedTV = function () {
+                Dashboard.navigate("livetv.html?tab=3");
+            };
+
+            appRouter.showFavorites = function () {
+                Dashboard.navigate("home.html?tab=1");
+            };
+
+            appRouter.showSettings = function () {
+                Dashboard.navigate("mypreferencesmenu.html");
+            };
+
+            appRouter.showNowPlaying = function () {
+                Dashboard.navigate("nowplaying.html");
+            };
+
+            appRouter.setTitle = function (title) {
+                LibraryMenu.setTitle(title);
+            };
+
+            appRouter.getRouteUrl = function (item, options) {
+                if (!item) {
+                    throw new Error("item cannot be null");
+                }
+
+                if (item.url) {
+                    return item.url;
+                }
+
+                var context = options ? options.context : null;
+                var id = item.Id || item.ItemId;
+
+                if (!options) {
+                    options = {};
+                }
+
+                var url;
+                var itemType = item.Type || (options ? options.itemType : null);
+                var serverId = item.ServerId || options.serverId;
+
+                if ("settings" === item) {
```

Comment:
We should probably convert all these to `switch` statements instead. It's out of the scope of this PR though.

Reply:
this could be converted to a switch but also moved to the module appRouter and not in site.js. I made a routes.js that removes all mappings from site.js but that is to be made in another PR. I dont remender if i already passed that code to appRouter module, but i could check later on.